File: Backend/app/dtos/users/user_response_dto.py
from app.dtos.users.user_auth_dto import UserRegisterDataDTO
from app.dtos.users.user_base_dto import UserResponseDataDTO, UserResponseDataDTOList
from app.dtos.common.base_response_dto import BaseResponseDTO
from app.dtos.users.user_action_dto import UserUpdateData


class UserRegisterResponseDTO(BaseResponseDTO[UserRegisterDataDTO]):
    pass

class UserLoginResponseDTO(UserRegisterResponseDTO):
    pass


class UserUpdateResponseDTO(BaseResponseDTO[UserUpdateData]):
    pass


class UserResponseDTO(BaseResponseDTO[UserResponseDataDTO]):
    pass

class UserResponseDTOList(BaseResponseDTO[UserResponseDataDTOList]):
    pass

File: Frontend/src/types/dtos/users/user_response.dto.ts
import type { ApiResponse } from "~/types/common/api-response.type";
import type { UserLoginDataDTO, UserRegisterDataDTO } from "./user_auth_dto";
import type { UserResponseDataDTO } from "./user_action_dto";

export type UserRegisterResponseDTO = ApiResponse<UserRegisterDataDTO>;

export type UserLoginResponseDTO = ApiResponse<UserLoginDataDTO>;

export type UserUpdateResponseDTO = ApiResponse<UserResponseDataDTO>;

export type UserResponseDTO = ApiResponse<UserResponseDataDTO>;

export type UserResponseDTOList = ApiResponse<UserResponseDataDTO[]>;
